.github/scripts/explain-sast-alerts.py

```
@@ -0,0 +1,287 @@
+
+def fetch_codeql_alerts(repo: str, pr_number: int) -> List[Dict[str, Any]]:
+    """Fetch CodeQL alerts for a pull request."""
+    headers = get_github_headers()
+    owner, repo_name = repo.split("/")
+    
+    # Get alerts from the PR's head branch
+    url = f"{GITHUB_API_BASE}/repos/{owner}/{repo_name}/code-scanning/alerts"
+    
+    # Get PR details to find the head branch
+    pr_url = f"{GITHUB_API_BASE}/repos/{owner}/{repo_name}/pulls/{pr_number}"
+    pr_response = requests.get(pr_url, headers=headers)
```

Ruff (0.14.7)

Probable use of `requests` call without timeout

(S113)